### PR TITLE
JBIDE-22691 - Allow extensions to be injected with Eclipse services

### PR DESCRIPTION
**Pull request by johnsteele**
It's common to need access to services from within extensions.

### CODE REVIEW

File: common/tests/org.jboss.tools.common.ui.test/src/org/jboss/tools/common/ext/test/E4ExtensionFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import org.eclipse.ui.plugin.AbstractUIPlugin;
+import org.jboss.tools.common.ext.E4ExtensionFactory;
+import org.jboss.tools.common.ui.test.Activator;
+import org.junit.Assert;
+import org.junit.Test;
+import org.osgi.framework.Bundle;
+
+public class E4ExtensionFactoryTest extends AbstractUIPlugin {
+	
+	public static class CommonExtensionFacotry extends E4ExtensionFactory {
```
Comment (fbricon) [inline common/tests/org.jboss.tools.common.ui.test/src/org/jboss/tools/common/ext/test/E4ExtensionFactoryTest.java]: typo

File: foundation/tests/org.jboss.tools.foundation.ui.test/src/org/jboss/tools/foundation/ui/test/Activator.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
+package org.jboss.tools.foundation.ui.test;
```
Comment (fbricon) [inline foundation/tests/org.jboss.tools.foundation.ui.test/src/org/jboss/tools/foundation/ui/test/Activator.java]: missing copyright header

**Comment by mickaelistria** [comment]
I don't get how the extension factories are actually registered? Especially since they are abstract, how does their behavior hook in the regular createExecutableExtension(...) ?

**Comment by mickaelistria** [comment]
Answer to myself: `org.jboss.tools.foundation.ui.test.ext.E4ExtensionFactoryTest$TestExtensionFactory:org.jboss.tools.foundation.ui.test.ext.TestView`

**Pull request merged**